extensions/amp-lightbox/1.0/amp-lightbox.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ActionTrust} from '../../../src/action-constants';
+import {Lightbox} from './lightbox';
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {dict} from '../../../src/utils/object';
+
+/** @const {string} */
+const TAG = 'amp-lightbox';
+
+class AmpLightbox extends PreactBaseElement {
+  /** @override */
+  init() {
+    this.registerApiAction('open', (api) => api.open(), ActionTrust.DEFAULT);
+    this.registerApiAction('close', (api) => api.close(), ActionTrust.DEFAULT);
+    return dict({'openOnLoad': false});
+  }
+}
+
+/** @override */
+AmpLightbox['Component'] = Lightbox;
+
+/** @override */
+AmpLightbox['props'] = {
+  'animateIn': {attr: 'animate-in'},
+  'scrollable': {attr: 'scrollable'},
+  'id': {attr: 'id'},
+  'layout': {attr: 'layout'},
+  'openOnLoad': {attr: 'openOnLoad'},
```

Comment:
Yup `openOnLoad` is a boolean. I'll make sure to change the name of it. One note though: this would technically result in an API addition/change compared to the 0.1 version. Is that okay?

Reply:
Yes. This is definitely not a mode we can allow in AMP b/c it will cause a massive CLS issue. I'm ok skipping it completely for now, unless there's something technically important to get the structure/APIs right. For instance, w/o this prop we wouldn't have noticed that animations are a problem on mount. If so, we can simply have it in Preact and remove it from AMP bindings.

Reply:
Leaving this conversation open for now as I want to discuss what we do with `open` from the previous API later on.